client/src/App.tsx

```diff
@@ -291,6 +295,43 @@ const App = () => {
     }
   };
 
+  const handleCompletion = async (
+    ref: ResourceReference | PromptReference,
+    argName: string,
+    value: string,
+    signal?: AbortSignal,
+  ) => {
+    if (!mcpClient) {
+      throw new Error("MCP client not connected");
+    }
+
+    const request: ClientRequest = {
+      method: "completion/complete",
+      params: {
+        argument: {
+          name: argName,
+          value,
+        },
+        ref,
+      },
+    };
+
+    try {
+      const response = await mcpClient.complete(request.params, {
+        signal,
+      });
+      pushHistory(request, response);
+
+      return response?.completion.values || [];
+    } catch (e: unknown) {
+      const errorMessage = e instanceof Error ? e.message : String(e);
+      pushHistory(request, { error: errorMessage });
+
+      toast.error(errorMessage);
+      throw e;
+    }
+  };
```

Comment:
Can we use the `makeRequest` function above?

Reply:
Sure thing!  Will work on this today.

As a side note - I've noticed that the current Spec does not allow servers to indicate whether they support completions, as far as I understand.  So if we request completions from a server that does not provide a handler, we get a Method not found error (-32601).  

In the short term, I can address this by tracking that specific error and disable further requests for the selected template or prompt.  

Long term I feel it would be good to include completions as part of the capability negotiation 
e.g. 
```typescript
const server = new Server({
  name: 'My server',
  version: '1.0.0',
}, {
    capabilities: {
      resources: { subscribe: true, completions: true },
      prompts: { completions: true},
    }
  })

```

Let me know your thoughts or if there's anything I might have overlooked.

Reply:
Fair callout. If you don't mind, could you open a discussion in the [specification repo](https://github.com/modelcontextprotocol/specification/discussions/111)?

Reply:
for sure - https://github.com/modelcontextprotocol/specification/discussions/122